strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/pypi/PypiArtifactController.java

```diff
@@ -0,0 +1,351 @@
+                                                                                  .setAuthorEmail(authorEmail)
+                                                                                  .setComment(comment)
+                                                                                  .setDescription(description)
+                                                                                  .setDownloadUrl(downloadUrl)
+                                                                                  .setFileType(filetype)
+                                                                                  .setHomePage(homePage)
+                                                                                  .setLicense(license)
+                                                                                  .setMd5Digest(md5Digest)
+                                                                                  .setMetdataVersion(metadataVersion)
+                                                                                  .setName(name)
+                                                                                  .setPlatform(platform)
+                                                                                  .setProtcolVersion(protcolVersion)
+                                                                                  .setPyVersion(pyversion)
+                                                                                  .setSummary(summary)
+                                                                                  .setVersion(version);
+
+            return validateAndUploadPackage(pypiArtifactMetadata, file, storageId, repositoryId);
+        }
+        catch (Exception e)
+        {
+            logger.error("Failed to process pypi upload request for storageId -> [{}] , repositoryId -> [{}]",
+                         storageId, repositoryId, e);
+
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+
+    }
+
+    @ApiOperation(value = "This Endpoint will be used for download/install pypi package")
+    @ApiResponses(value = { @ApiResponse(code = HttpURLConnection.HTTP_OK, message = "Success"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_SEE_OTHER, message = "See Other Location"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_NOT_FOUND, message = "Request Url Not Found"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_INTERNAL_ERROR, message = "An error occurred while executing download request."),
+                            @ApiResponse(code = HttpURLConnection.HTTP_UNAVAILABLE, message = "Service Unavailable.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(path = "{storageId}/{repositoryId}/{packageName}/", method = RequestMethod.GET)
```

Comment:
should it be `/{storageId}/{repositoryId}/{packageName}` instead?

Reply:
This is the expected path  `/storages/storage-pypi/pypi-releases/pip-package-with-dependency/`

Reply:
I mean that currently there is no beginning slash and end slash seems redundant as well.